src/components/tl/mlx5/mcast/tl_mlx5_mcast_rcache.c

```diff
@@ -0,0 +1,161 @@
+    region  = ucc_derived_of(rregion, ucc_tl_mlx5_mcast_rcache_region_t);
+
+    ret     = ucc_tl_mlx5_mcast_coll_reg_mr((ucc_tl_mlx5_mcast_coll_context_t *)context, address,
+                                            length, &region->reg.mr);
+    if (ret < 0) {
+        return UCS_ERR_INVALID_PARAM;
+    } else {
+        return UCS_OK;
+    }
+}
+
+static void ucc_tl_mlx5_mcast_rcache_mem_dereg_cb(void *context, ucc_rcache_t *rcache,
+                                                  ucc_rcache_region_t *rregion)
+{
+    ucc_tl_mlx5_mcast_rcache_region_t *region = ucc_derived_of(rregion,
+                                                ucc_tl_mlx5_mcast_rcache_region_t);
+
+    ucc_tl_mlx5_mcast_coll_dereg_mr((ucc_tl_mlx5_mcast_coll_context_t *)context, region->reg.mr);
+}
+
+static void ucc_tl_mlx5_mcast_rcache_dump_region_cb(void *context, ucc_rcache_t *rcache,
+                                                    ucc_rcache_region_t *rregion, char *buf,
+                                                    size_t max)
+{
+    ucc_tl_mlx5_mcast_rcache_region_t *region = ucc_derived_of(rregion,
+                                           ucc_tl_mlx5_mcast_rcache_region_t);
+
+    snprintf(buf, max, "bar ptr:%p", region->reg.mr);
+}
+
+
+ucc_status_t
+ucc_tl_mlx5_mcast_mem_register(ucc_tl_mlx5_mcast_coll_context_t *ctx, void *addr,
+                               size_t length, ucc_tl_mlx5_mcast_reg_t **reg)
+{
+    ucc_rcache_region_t               *rregion;
+    ucc_tl_mlx5_mcast_rcache_region_t *region;
+    ucc_status_t                       status;
+    ucc_rcache_t                      *rcache;
+
+    rcache = ctx->rcache;
+
+    assert(rcache != NULL);
+
+    status = ucc_rcache_get(rcache, (void *)addr, length, NULL,
+                        &rregion);
+    if (ucc_unlikely(UCC_OK != status)) {
+        tl_error(ctx->lib, "ucc_rcache_get failed\n");
+        return UCC_ERR_INVALID_PARAM;
+    }
+
+    region = ucc_derived_of(rregion, ucc_tl_mlx5_mcast_rcache_region_t);
+    *reg   = &region->reg;
+
+    tl_debug(ctx->lib, "Memory register mr %p", (*reg)->mr);
+
+    return UCC_OK;
+}
+
+ucc_status_t
+ucc_tl_mlx5_mcast_mem_deregister(ucc_tl_mlx5_mcast_coll_context_t *ctx, ucc_tl_mlx5_mcast_reg_t *reg)
+{
+    ucc_tl_mlx5_mcast_rcache_region_t *region;
+    ucc_rcache_t                      *rcache;
+
+    rcache = ctx->rcache;
+
+    if (reg == NULL) return UCC_OK;
+
+    assert(rcache != NULL);
+
+    tl_debug(ctx->lib, "Memory deregister mr %p", reg->mr);
+
+    region = ucc_container_of(reg, ucc_tl_mlx5_mcast_rcache_region_t, reg);
+
+    ucc_rcache_region_put(rcache, &region->super);
+
+    return UCC_OK;
+}
+
+ucc_rcache_ops_t ucc_tl_mlx5_mcast_rcache_ops = {
+    .mem_reg     = ucc_tl_mlx5_mcast_rcache_mem_reg_cb,
+    .mem_dereg   = ucc_tl_mlx5_mcast_rcache_mem_dereg_cb,
+    .dump_region = ucc_tl_mlx5_mcast_rcache_dump_region_cb
+};
+
+ucc_status_t ucc_tl_mlx5_mcast_setup_rcache(ucc_tl_mlx5_mcast_coll_context_t *ctx)
```

Comment:
I suggest the name `ucc_tl_mlx5_mcast_rcache_create` to match current convention